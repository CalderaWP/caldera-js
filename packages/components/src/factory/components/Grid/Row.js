import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Flex, Box } from '@rebass/grid';
import { Column } from './Column';
/**
 *
 * @param columns
 * @param onChange
 * @param onBlur
 * @param className
 * @param rowId
 * @param children
 * @return {*}
 * @constructor
 */
export const Row = ({
	columns,
	onChange,
	onBlur,
	className,
	rowId,
	children
}) => {
	return (
		<Flex className={classNames('caldera-row', className)} id={rowId}>
			{columns ? (
				<Fragment>
					{columns.map(column => {
						let { width, padding, columnId } = column;

						return (
							<Column
								key={columnId}
								id={columnId}
								width={width}
								px={padding}
								py={padding}
							>
								{children}
							</Column>
						);
					})}
				</Fragment>
			) : (
				<Fragment>{children}</Fragment>
			)}
		</Flex>
	);
};

/**
 * Row prop types
 *
 * @type {{columns: shim, rowId: *, onChange: *, onBlur: shim, className: shim}}
 */
export const rowPropTypes = {
	columns: PropTypes.arrayOf(
		PropTypes.shape({
			width: PropTypes.string.isRequired,
			columnId: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
		})
	),
	rowId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
	onChange: PropTypes.func,
	onBlur: PropTypes.func,
	className: PropTypes.string
};
Row.propTypes = rowPropTypes;
