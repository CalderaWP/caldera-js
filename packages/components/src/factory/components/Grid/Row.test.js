import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import {
	checkboxFieldset,
	selectField,
	checkboxField,
	numberField,
	textField,
	emailField
} from '../../fields.fixtures';

import { Row } from './Row';

describe('Row component', () => {
	let onChange = jest.fn();
	let onBlur;
	beforeEach(() => {
		onChange = jest.fn();
		onBlur = jest.fn();
	});

	const columns = [
		{
			children: <div>Half Column</div>,
			width: '1/2',
			columnId: 1
		},
		{
			children: <div>Quarter Column</div>,
			width: '1/4',
			columnId: 2
		},
		{
			children: <div>Quarter Column 2</div>,
			width: '1/4',
			columnId: 3
		}
	];

	it('Creates columns', () => {
		const component = renderer.create(
			<Row
				rowId={'twoColumns'}
				columns={columns}
				onChange={onChange}
				onBlur={onBlur}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Renders children if no columns', () => {
		const component = shallow(
			<Row rowId={'twoColumns'} onChange={onChange} onBlur={onBlur}>
				<p>Hi Roy</p>
			</Row>
		);
		expect(component.find('p').text()).toEqual('Hi Roy');
	});

	it('Applies padding to x and y ', () => {
		const component = renderer.create(
			<Row
				rowId={'twoColumns'}
				columns={[
					{
						children: <div>Half Column</div>,
						width: '1/2',
						columnId: 1,
						padding: '222'
					}
				]}
				onChange={onChange}
				onBlur={onBlur}
				padding={'11'}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
