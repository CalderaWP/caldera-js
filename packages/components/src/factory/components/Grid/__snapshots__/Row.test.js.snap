// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component Applies padding to x and y  1`] = `
<div
  className="sc-bdVaJa caldera-row sc-bwzfXH pjOvZ"
  id="twoColumns"
>
  <div
    className="caldera-column width-6 sc-bdVaJa gfmZBt"
    width="1/2"
  />
</div>
`;

exports[`Row component Creates columns 1`] = `
<div
  className="sc-bdVaJa caldera-row sc-bwzfXH pjOvZ"
  id="twoColumns"
>
  <div
    className="caldera-column width-6 sc-bdVaJa gfmZBt"
    width="1/2"
  />
  <div
    className="caldera-column width-4 sc-bdVaJa jlBFAP"
    width="1/4"
  />
  <div
    className="caldera-column width-4 sc-bdVaJa jlBFAP"
    width="1/4"
  />
</div>
`;
